hyperspy/_signals/lazy.py

```diff
@@ -162,8 +162,10 @@ def close_file(self):
                 _logger.exception("Failed to close lazy Signal file")
 
     def _clear_temp_dask_data(self, obj=None):
-        del self._temp_dask_chunk
-        del self._temp_dask_chunk_slice
+        if hasattr(self, '_temp_dask_chunk'):
```

Comment:
Just curious - would it be better to have `try` here instead?

Reply:
That would work as well, but not sure what is the "recommended" way of doing these kinds of things.

Reply:
Looks like there are just different schools of thought on this, so this is fine.